exercises/practice/sum-of-multiples/.approaches/filter-for-multiples/content.md

```diff
@@ -0,0 +1,253 @@
+
+1. look at every individual integer between `0` and `limit`,
+2. check that it is a multiple of any of the given `factors`, and
+3. add it to the sum when it is.
+
+
+## Notable language features used in this solution
+
+### Built-in function: `sum`
+
+Adding all the numbers in a collection together is a very common operation.
+Therefore, Python provides the built-in function [`sum`][builtin-sum].
+
+`sum` takes one argument, and requires that it be **iterable**.
+A value is iterable whenever it makes sense to use it in a `for` loop like this:
+
+```python
+for element in iterable_value:  # 👈
+    ...
+```
+
+The `list` is the most commonly used iterable data structure.
+Many other containers are also iterable, such as `set`s, `tuple`s, `range`s, and even `dict`s and `str`ings.
+Still other examples include iterators and generators, which are discussed below.
+
+When given a collection of numbers, `sum` will look at the elements one by one and add them up.
+The result is a single number.
+
+```python
+numbers = range(1, 100 + 1)  # 1, 2, …, 100
+sum(numbers)  # ⟹ 5050
+```
+
+Had the highlighted solution not used `sum`, it might have looked like this:
+
+```python
+def sum_of_multiples(limit, factors):
+    multiples = filter(
+        lambda n: any(n % f == 0 for f in factors if f != 0),
+        range(limit))
+    total = 0
+    for multiple in multiples:
+        total += multiple
+    return total
+```
+
+
+### Built-in function: `filter`
+
+Selecting elements of a collection for having a certain property is also a very common operation.
+Therefore, Python provides the built-in function [`filter`][builtin-filter].
+
+`filter` takes two arguments.
+The first is a **predicate**.
+The second is the iterable the elements of which should be filtered.
+
+A predicate is a function that takes one argument (of any particular type) and returns a `bool`.
+Such functions are commonly used to encode properties of values.
+An example is `str.isupper`, which takes a `str` and returns `True` whenever it is uppercase:
+
+```python
+str.isupper("AAAAH! 😱")  # ⟹ True
+str.isupper("Eh? 😕")     # ⟹ False
+str.isupper("⬆️💼")       # ⟹ False
+```
+
+Thus, the function `str.isupper` represents the property of _being an uppercase string_.
+
+Contrary to what you might expect, `filter` does not return a data structure like the one given as the iterable argument:
+
+```python
+filter(str.isupper, ["THUNDERBOLTS", "and", "LIGHTNING"])
+# ⟹ <filter object at 0x000002F46B107BE0>
+```
+
+Instead, it returns an **iterator**.
+
+An iterator is an object whose sole purpose is to guide iteration through some data structure.
+In particular, `filter` makes sure that elements that do not satisfy the predicate are skipped:
+
+```python
+for word in filter(str.isupper, ["THUNDERBOLTS", "and", "LIGHTNING"]):
+    print(word)
+# prints:
+# THUNDERBOLTS
+# LIGHTNING
+```
+
+An iterator is a bit like a cursor that can move only to the right.
```

Comment:
I really like this explanation.  Both from a cursor for typing ... and a cursor from a DB.  Both can only be gone through once, and you can't back up.  😄